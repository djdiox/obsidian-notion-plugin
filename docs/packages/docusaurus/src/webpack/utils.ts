/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import fs from 'fs-extra';
import path from 'path';
import crypto from 'crypto';
import logger from '@docusaurus/logger';
import {BABEL_CONFIG_FILE_NAME} from '@docusaurus/utils';
import MiniCssExtractPlugin from 'mini-css-extract-plugin';
import {
  mergeWithCustomize,
  customizeArray,
  customizeObject,
} from 'webpack-merge';
import webpack, {
  type Configuration,
  type RuleSetRule,
  type WebpackPluginInstance,
} from 'webpack';
import TerserPlugin from 'terser-webpack-plugin';
import CssMinimizerPlugin from 'css-minimizer-webpack-plugin';
import type {CustomOptions, CssNanoOptions} from 'css-minimizer-webpack-plugin';
import type {TransformOptions} from '@babel/core';
import type {
  Plugin,
  PostCssOptions,
  ConfigureWebpackUtils,
} from '@docusaurus/types';

// Utility method to get style loaders
export function getStyleLoaders(
  isServer: boolean,
  cssOptionsArg: {
    [key: string]: unknown;
  } = {},
): RuleSetRule[] {
  const cssOptions: {[key: string]: unknown} = {
    // TODO turn esModule on later, see https://github.com/facebook/docusaurus/pull/6424
    esModule: false,
    ...cssOptionsArg,
  };

  if (isServer) {
    return cssOptions.modules
      ? [
          {
            loader: require.resolve('css-loader'),
            options: cssOptions,
          },
        ]
      : [
          {
            loader: MiniCssExtractPlugin.loader,
            options: {
              // Don't emit CSS files for SSR (previously used null-loader)
              // See https://github.com/webpack-contrib/mini-css-extract-plugin/issues/90#issuecomment-811991738
              emit: false,
            },
          },
          {
            loader: require.resolve('css-loader'),
            options: cssOptions,
          },
        ];
  }

  return [
    {
      loader: MiniCssExtractPlugin.loader,
      options: {
        esModule: true,
      },
    },
    {
      loader: require.resolve('css-loader'),
      options: cssOptions,
    },
    {
      // Options for PostCSS as we reference these options twice
      // Adds vendor prefixing based on your specified browser support in
      // package.json
      loader: require.resolve('postcss-loader'),
      options: {
        postcssOptions: {
          // Necessary for external CSS imports to work
          // https://github.com/facebook/create-react-app/issues/2677
          ident: 'postcss',
          plugins: [
            // eslint-disable-next-line global-require
            require('autoprefixer'),
          ],
        },
      },
    },
  ];
}

export async function getCustomBabelConfigFilePath(
  siteDir: string,
): Promise<string | undefined> {
  const customBabelConfigurationPath = path.join(
    siteDir,
    BABEL_CONFIG_FILE_NAME,
  );
  return (await fs.pathExists(customBabelConfigurationPath))
    ? customBabelConfigurationPath
    : undefined;
}

export function getBabelOptions({
  isServer,
  babelOptions,
}: {
  isServer?: boolean;
  babelOptions?: TransformOptions | string;
} = {}): TransformOptions {
  if (typeof babelOptions === 'string') {
    return {
      babelrc: false,
      configFile: babelOptions,
      caller: {name: isServer ? 'server' : 'client'},
    };
  }
  return {
    ...(babelOptions ?? {presets: [require.resolve('../babel/preset')]}),
    babelrc: false,
    configFile: false,
    caller: {name: isServer ? 'server' : 'client'},
  };
}

// Name is generic on purpose
// we want to support multiple js loader implementations (babel + esbuild)
function getDefaultBabelLoader({
  isServer,
  babelOptions,
}: {
  isServer: boolean;
  babelOptions?: TransformOptions | string;
}): RuleSetRule {
  return {
    loader: require.resolve('babel-loader'),
    options: getBabelOptions({isServer, babelOptions}),
  };
}

export const getCustomizableJSLoader =
  (jsLoader: 'babel' | ((isServer: boolean) => RuleSetRule) = 'babel') =>
  ({
    isServer,
    babelOptions,
  }: {
    isServer: boolean;
    babelOptions?: TransformOptions | string;
  }): RuleSetRule =>
    jsLoader === 'babel'
      ? getDefaultBabelLoader({isServer, babelOptions})
      : jsLoader(isServer);

/**
 * Helper function to modify webpack config
 * @param configureWebpack a webpack config or a function to modify config
 * @param config initial webpack config
 * @param isServer indicates if this is a server webpack configuration
 * @param jsLoader custom js loader config
 * @param content content loaded by the plugin
 * @returns final/ modified webpack config
 */
export function applyConfigureWebpack(
  configureWebpack: NonNullable<Plugin['configureWebpack']>,
  config: Configuration,
  isServer: boolean,
  jsLoader: 'babel' | ((isServer: boolean) => RuleSetRule) | undefined,
  content: unknown,
): Configuration {
  // Export some utility functions
  const utils: ConfigureWebpackUtils = {
    getStyleLoaders,
    getJSLoader: getCustomizableJSLoader(jsLoader),
  };
  if (typeof configureWebpack === 'function') {
    const {mergeStrategy, ...res} =
      configureWebpack(config, isServer, utils, content) ?? {};
    const customizeRules = mergeStrategy ?? {};
    return mergeWithCustomize({
      customizeArray: customizeArray(customizeRules),
      customizeObject: customizeObject(customizeRules),
    })(config, res);
  }
  return config;
}

export function applyConfigurePostCss(
  configurePostCss: NonNullable<Plugin['configurePostCss']>,
  config: Configuration,
): Configuration {
  type LocalPostCSSLoader = object & {
    options: {postcssOptions: PostCssOptions};
  };

  // Not ideal heuristic but good enough for our use-case?
  function isPostCssLoader(loader: unknown): loader is LocalPostCSSLoader {
    return !!(loader as LocalPostCSSLoader)?.options?.postcssOptions;
  }

  // Does not handle all edge cases, but good enough for now
  function overridePostCssOptions(entry: RuleSetRule) {
    if (isPostCssLoader(entry)) {
      entry.options.postcssOptions = configurePostCss(
        entry.options.postcssOptions,
      );
    } else if (Array.isArray(entry.oneOf)) {
      entry.oneOf.forEach(overridePostCssOptions);
    } else if (Array.isArray(entry.use)) {
      entry.use
        .filter((u) => typeof u === 'object')
        .forEach((rule) => overridePostCssOptions(rule as RuleSetRule));
    }
  }

  config.module?.rules?.forEach((rule) =>
    overridePostCssOptions(rule as RuleSetRule),
  );

  return config;
}

declare global {
  interface Error {
    /** @see https://webpack.js.org/api/node/#error-handling */
    details: unknown;
  }
}

export function compile(config: Configuration[]): Promise<void> {
  return new Promise((resolve, reject) => {
    const compiler = webpack(config);
    compiler.run((err, stats) => {
      if (err) {
        logger.error(err.stack ?? err);
        if (err.details) {
          logger.error(err.details);
        }
        reject(err);
      }
      // Let plugins consume all the stats
      const errorsWarnings = stats?.toJson('errors-warnings');
      if (stats?.hasErrors()) {
        reject(new Error('Failed to compile with errors.'));
      }
      if (errorsWarnings && stats?.hasWarnings()) {
        errorsWarnings.warnings?.forEach((warning) => {
          logger.warn(warning);
        });
      }
      // Webpack 5 requires calling close() so that persistent caching works
      // See https://github.com/webpack/webpack.js.org/pull/4775
      compiler.close((errClose) => {
        if (errClose) {
          logger.error(`Error while closing Webpack compiler: ${errClose}`);
          reject(errClose);
        } else {
          resolve();
        }
      });
    });
  });
}

// Ensure the certificate and key provided are valid and if not
// throw an easy to debug error
function validateKeyAndCerts({
  cert,
  key,
  keyFile,
  crtFile,
}: {
  cert: Buffer;
  key: Buffer;
  keyFile: string;
  crtFile: string;
}) {
  let encrypted: Buffer;
  try {
    // publicEncrypt will throw an error with an invalid cert
    encrypted = crypto.publicEncrypt(cert, Buffer.from('test'));
  } catch (err) {
    logger.error`The certificate path=${crtFile} is invalid.`;
    throw err;
  }

  try {
    // privateDecrypt will throw an error with an invalid key
    crypto.privateDecrypt(key, encrypted);
  } catch (err) {
    logger.error`The certificate key path=${keyFile} is invalid.`;
    throw err;
  }
}

// Read file and throw an error if it doesn't exist
async function readEnvFile(file: string, type: string) {
  if (!(await fs.pathExists(file))) {
    throw new Error(
      `You specified ${type} in your env, but the file "${file}" can't be found.`,
    );
  }
  return fs.readFile(file);
}

// Get the https config
// Return cert files if provided in env, otherwise just true or false
export async function getHttpsConfig(): Promise<
  boolean | {cert: Buffer; key: Buffer}
> {
  const appDirectory = await fs.realpath(process.cwd());
  const {SSL_CRT_FILE, SSL_KEY_FILE, HTTPS} = process.env;
  const isHttps = HTTPS === 'true';

  if (isHttps && SSL_CRT_FILE && SSL_KEY_FILE) {
    const crtFile = path.resolve(appDirectory, SSL_CRT_FILE);
    const keyFile = path.resolve(appDirectory, SSL_KEY_FILE);
    const config = {
      cert: await readEnvFile(crtFile, 'SSL_CRT_FILE'),
      key: await readEnvFile(keyFile, 'SSL_KEY_FILE'),
    };

    validateKeyAndCerts({...config, keyFile, crtFile});
    return config;
  }
  return isHttps;
}

// See https://github.com/webpack-contrib/terser-webpack-plugin#parallel
function getTerserParallel() {
  let terserParallel: boolean | number = true;
  if (process.env.TERSER_PARALLEL === 'false') {
    terserParallel = false;
  } else if (
    process.env.TERSER_PARALLEL &&
    parseInt(process.env.TERSER_PARALLEL, 10) > 0
  ) {
    terserParallel = parseInt(process.env.TERSER_PARALLEL, 10);
  }
  return terserParallel;
}

export function getMinimizer(
  useSimpleCssMinifier = false,
): WebpackPluginInstance[] {
  const minimizer: WebpackPluginInstance[] = [
    new TerserPlugin({
      parallel: getTerserParallel(),
      terserOptions: {
        parse: {
          // We want uglify-js to parse ecma 8 code. However, we don't want it
          // to apply any minification steps that turns valid ecma 5 code
          // into invalid ecma 5 code. This is why the 'compress' and 'output'
          // sections only apply transformations that are ecma 5 safe
          // https://github.com/facebook/create-react-app/pull/4234
          ecma: 2020,
        },
        compress: {
          ecma: 5,
          warnings: false,
        },
        mangle: {
          safari10: true,
        },
        output: {
          ecma: 5,
          comments: false,
          // Turned on because emoji and regex is not minified properly using
          // default. See https://github.com/facebook/create-react-app/issues/2488
          ascii_only: true,
        },
      },
    }),
  ];
  if (useSimpleCssMinifier) {
    minimizer.push(new CssMinimizerPlugin());
  } else {
    minimizer.push(
      // Using the array syntax to add 2 minimizers
      // see https://github.com/webpack-contrib/css-minimizer-webpack-plugin#array
      new CssMinimizerPlugin<[CssNanoOptions, CustomOptions]>({
        minimizerOptions: [
          // CssNano options
          {
            preset: require.resolve('@docusaurus/cssnano-preset'),
          },
          // CleanCss options
          {
            inline: false,
            level: {
              1: {
                all: false,
                removeWhitespace: true,
              },
              2: {
                all: true,
                restructureRules: true,
                removeUnusedAtRules: false,
              },
            },
          },
        ],
        minify: [
          CssMinimizerPlugin.cssnanoMinify,
          CssMinimizerPlugin.cleanCssMinify,
        ],
      }),
    );
  }

  return minimizer;
}
