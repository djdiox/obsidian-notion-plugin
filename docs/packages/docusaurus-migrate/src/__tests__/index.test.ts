/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {jest} from '@jest/globals';
import path from 'path';
import fs from 'fs-extra';
import {posixPath} from '@docusaurus/utils';
import {migrateDocusaurusProject} from '../index';

async function testMigration(siteDir: string, newDir: string) {
  const writeMock = jest.spyOn(fs, 'outputFile').mockImplementation(() => {});
  const mkdirpMock = jest.spyOn(fs, 'mkdirp').mockImplementation(() => {});
  const mkdirsMock = jest.spyOn(fs, 'mkdirs').mockImplementation(() => {});
  const copyMock = jest.spyOn(fs, 'copy').mockImplementation(() => {});
  await migrateDocusaurusProject(siteDir, newDir, true, true);
  expect(
    writeMock.mock.calls.sort((a, b) =>
      posixPath(a[0]).localeCompare(posixPath(b[0])),
    ),
  ).toMatchSnapshot('write');
  expect(
    mkdirpMock.mock.calls.sort((a, b) =>
      posixPath(a[0]).localeCompare(posixPath(b[0])),
    ),
  ).toMatchSnapshot('mkdirp');
  expect(
    mkdirsMock.mock.calls.sort((a, b) =>
      posixPath(a[0]).localeCompare(posixPath(b[0])),
    ),
  ).toMatchSnapshot('mkdirs');
  expect(
    copyMock.mock.calls.sort((a, b) =>
      posixPath(a[0]).localeCompare(posixPath(b[0])),
    ),
  ).toMatchSnapshot('copy');
  writeMock.mockRestore();
  mkdirpMock.mockRestore();
  mkdirsMock.mockRestore();
  copyMock.mockRestore();
}

describe('migration CLI', () => {
  const fixtureDir = path.join(__dirname, '__fixtures__');
  it('migrates simple website', async () => {
    const siteDir = path.join(fixtureDir, 'simple_website', 'website');
    const newDir = path.join(fixtureDir, 'migrated_simple_site');
    await testMigration(siteDir, newDir);
  });

  it('migrates complex website', async () => {
    const siteDir = path.join(fixtureDir, 'complex_website', 'website');
    const newDir = path.join(fixtureDir, 'migrated_complex_site');
    await testMigration(siteDir, newDir);
  });

  it('migrates missing versions', async () => {
    const siteDir = path.join(fixtureDir, 'missing_version_website', 'website');
    const newDir = path.join(fixtureDir, 'migrated_missing_version_site');
    await testMigration(siteDir, newDir);
  });
});
