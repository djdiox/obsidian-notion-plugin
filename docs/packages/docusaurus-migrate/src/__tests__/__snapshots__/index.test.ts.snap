// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migration CLI migrates complex website: copy 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/complex_website/website/blog",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/blog",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/complex_website/website/pages/en",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/src/pages",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/complex_website/website/static",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/static",
  ],
]
`;

exports[`migration CLI migrates complex website: mkdirp 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/src/pages",
  ],
]
`;

exports[`migration CLI migrates complex website: mkdirs 1`] = `[]`;

exports[`migration CLI migrates complex website: write 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/docusaurus.config.js",
    "module.exports={
  "title": "Docusaurus",
  "tagline": "Easy to Maintain Open Source Documentation Websites",
  "url": "https://docusaurus.io",
  "baseUrl": "/",
  "organizationName": "facebook",
  "projectName": "docusaurus",
  "noIndex": false,
  "scripts": [
    "https://buttons.github.io/buttons.js",
    "https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js",
    "/js/code-blocks-buttons.js"
  ],
  "favicon": "img/docusaurus.ico",
  "customFields": {
    "users": {
      "caption": "DevSpace",
      "image": "/img/users/devspace.svg",
      "infoLink": "https://devspace.cloud/docs/",
      "fbOpenSource": false,
      "pinned": false
    },
    "translationRecruitingLink": "https://crowdin.com/project/docusaurus",
    "facebookAppId": "199138890728411"
  },
  "onBrokenLinks": "log",
  "onBrokenMarkdownLinks": "log",
  "presets": [
    [
      "@docusaurus/preset-classic",
      {
        "docs": {
          "showLastUpdateAuthor": true,
          "showLastUpdateTime": true,
          "editUrl": "https://github.com/facebook/docusaurus/edit/main/docs/"
        },
        "blog": {},
        "theme": {
          "customCss": "../complex_website/src/css/customTheme.css"
        },
        "googleAnalytics": {
          "trackingID": "UA-44373548-31"
        }
      }
    ]
  ],
  "plugins": [],
  "themeConfig": {
    "navbar": {
      "title": "Docusaurus",
      "logo": {
        "src": "img/docusaurus.svg"
      },
      "items": [
        {
          "to": "docs/installation",
          "label": "Docs",
          "position": "left"
        },
        {
          "to": "docs/tutorial-setup",
          "label": "Tutorial",
          "position": "left"
        },
        {
          "to": "/users",
          "label": "Users",
          "position": "left"
        },
        {
          "href": "https://github.com/facebook/docusaurus",
          "label": "GitHub",
          "position": "left"
        }
      ]
    },
    "image": "img/docusaurus.png",
    "footer": {
      "links": [
        {
          "title": "Community",
          "items": [
            {
              "label": "Twitter",
              "to": "https://twitter.com/docusaurus"
            }
          ]
        }
      ],
      "copyright": "Copyright © 2022 Facebook Inc.",
      "logo": {
        "src": "img/docusaurus_monochrome.svg"
      }
    },
    "algolia": {
      "apiKey": "3eb9507824b8be89e7a199ecaa1a9d2c",
      "indexName": "docusaurus",
      "algoliaOptions": {
        "facetFilters": [
          "language:LANGUAGE",
          "version:VERSION"
        ]
      }
    }
  }
}",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/package.json",
    "{
  "name": "docusaurus-1-website",
  "version": "2.0.0-alpha.58",
  "private": true,
  "scripts": {
    "start": "docusaurus start",
    "build": "docusaurus build",
    "publish-gh-pages": "docusaurus-publish",
    "examples": "docusaurus-examples",
    "write-translations": "docusaurus-write-translations",
    "docusaurus-version": "docusaurus-version",
    "rename-version": "docusaurus-rename-version",
    "crowdin-upload": "crowdin --config ../crowdin.yaml upload sources --auto-update -b master",
    "crowdin-download": "crowdin --config ../crowdin.yaml download -b master",
    "swizzle": "docusaurus swizzle",
    "deploy": "docusaurus deploy",
    "docusaurus": "docusaurus"
  },
  "dependencies": {
    "@docusaurus/core": "<CURRENT_VERSION>",
    "@docusaurus/preset-classic": "<CURRENT_VERSION>",
    "clsx": "^1.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_complex_site/src/css/customTheme.css",
    ":root{
  --ifm-color-primary-lightest: #3CAD6E;
  --ifm-color-primary-lighter: #359962;
  --ifm-color-primary-light: #33925D;
  --ifm-color-primary: #2E8555;
  --ifm-color-primary-dark: #29784C;
  --ifm-color-primary-darker: #277148;
  --ifm-color-primary-darkest: #205D3B;
}
",
  ],
]
`;

exports[`migration CLI migrates missing versions: copy 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/missing_version_website/website/blog",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/blog",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/missing_version_website/website/pages/en",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/src/pages",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/missing_version_website/website/static",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/static",
  ],
]
`;

exports[`migration CLI migrates missing versions: mkdirp 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/src/pages",
  ],
]
`;

exports[`migration CLI migrates missing versions: mkdirs 1`] = `[]`;

exports[`migration CLI migrates missing versions: write 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/docusaurus.config.js",
    "module.exports={
  "title": "Docusaurus",
  "tagline": "Easy to Maintain Open Source Documentation Websites",
  "url": "https://docusaurus.io",
  "baseUrl": "/",
  "organizationName": "facebook",
  "projectName": "docusaurus",
  "noIndex": false,
  "scripts": [
    "https://buttons.github.io/buttons.js",
    "https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js",
    "/js/code-blocks-buttons.js"
  ],
  "favicon": "img/docusaurus.ico",
  "customFields": {
    "users": {
      "caption": "DevSpace",
      "image": "/img/users/devspace.svg",
      "infoLink": "https://devspace.cloud/docs/",
      "fbOpenSource": false,
      "pinned": false
    },
    "translationRecruitingLink": "https://crowdin.com/project/docusaurus",
    "facebookAppId": "199138890728411"
  },
  "onBrokenLinks": "log",
  "onBrokenMarkdownLinks": "log",
  "presets": [
    [
      "@docusaurus/preset-classic",
      {
        "docs": {
          "showLastUpdateAuthor": true,
          "showLastUpdateTime": true,
          "editUrl": "https://github.com/facebook/docusaurus/edit/main/docs/"
        },
        "blog": {},
        "theme": {
          "customCss": "../missing_version_website/src/css/customTheme.css"
        },
        "googleAnalytics": {
          "trackingID": "UA-44373548-31"
        }
      }
    ]
  ],
  "plugins": [],
  "themeConfig": {
    "navbar": {
      "title": "Docusaurus",
      "logo": {
        "src": "img/docusaurus.svg"
      },
      "items": [
        {
          "to": "docs/installation",
          "label": "Docs",
          "position": "left"
        },
        {
          "to": "docs/tutorial-setup",
          "label": "Tutorial",
          "position": "left"
        },
        {
          "to": "/users",
          "label": "Users",
          "position": "left"
        },
        {
          "href": "https://github.com/facebook/docusaurus",
          "label": "GitHub",
          "position": "left"
        }
      ]
    },
    "image": "img/docusaurus.png",
    "footer": {
      "links": [
        {
          "title": "Community",
          "items": [
            {
              "label": "Twitter",
              "to": "https://twitter.com/docusaurus"
            }
          ]
        }
      ],
      "copyright": "Copyright © 2022 Facebook Inc.",
      "logo": {
        "src": "img/docusaurus_monochrome.svg"
      }
    },
    "algolia": {
      "apiKey": "3eb9507824b8be89e7a199ecaa1a9d2c",
      "indexName": "docusaurus",
      "algoliaOptions": {
        "facetFilters": [
          "language:LANGUAGE",
          "version:VERSION"
        ]
      }
    }
  }
}",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/package.json",
    "{
  "name": "docusaurus-1-website",
  "version": "2.0.0-alpha.58",
  "private": true,
  "scripts": {
    "start": "docusaurus start",
    "build": "docusaurus build",
    "publish-gh-pages": "docusaurus-publish",
    "examples": "docusaurus-examples",
    "write-translations": "docusaurus-write-translations",
    "docusaurus-version": "docusaurus-version",
    "rename-version": "docusaurus-rename-version",
    "crowdin-upload": "crowdin --config ../crowdin.yaml upload sources --auto-update -b master",
    "crowdin-download": "crowdin --config ../crowdin.yaml download -b master",
    "swizzle": "docusaurus swizzle",
    "deploy": "docusaurus deploy",
    "docusaurus": "docusaurus"
  },
  "dependencies": {
    "@docusaurus/core": "<CURRENT_VERSION>",
    "@docusaurus/preset-classic": "<CURRENT_VERSION>",
    "clsx": "^1.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_missing_version_site/src/css/customTheme.css",
    ":root{
  --ifm-color-primary-lightest: #3CAD6E;
  --ifm-color-primary-lighter: #359962;
  --ifm-color-primary-light: #33925D;
  --ifm-color-primary: #2E8555;
  --ifm-color-primary-dark: #29784C;
  --ifm-color-primary-darker: #277148;
  --ifm-color-primary-darkest: #205D3B;
}
",
  ],
]
`;

exports[`migration CLI migrates simple website: copy 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/simple_website/website/pages/en",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/src/pages",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/simple_website/website/static",
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/static",
  ],
]
`;

exports[`migration CLI migrates simple website: mkdirp 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/src/pages",
  ],
]
`;

exports[`migration CLI migrates simple website: mkdirs 1`] = `[]`;

exports[`migration CLI migrates simple website: write 1`] = `
[
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/docusaurus.config.js",
    "module.exports={
  "title": "Docusaurus",
  "tagline": "Easy to Maintain Open Source Documentation Websites",
  "url": "https://docusaurus.io",
  "baseUrl": "/",
  "organizationName": "facebook",
  "projectName": "docusaurus",
  "noIndex": false,
  "scripts": [
    "https://buttons.github.io/buttons.js",
    "https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js",
    "/js/code-blocks-buttons.js"
  ],
  "favicon": "img/docusaurus.ico",
  "customFields": {
    "users": {
      "caption": "DevSpace",
      "image": "/img/users/devspace.svg",
      "infoLink": "https://devspace.cloud/docs/",
      "fbOpenSource": false,
      "pinned": false
    },
    "translationRecruitingLink": "https://crowdin.com/project/docusaurus",
    "facebookAppId": "199138890728411"
  },
  "onBrokenLinks": "log",
  "onBrokenMarkdownLinks": "log",
  "presets": [
    [
      "@docusaurus/preset-classic",
      {
        "docs": {
          "showLastUpdateAuthor": true,
          "showLastUpdateTime": true,
          "editUrl": "https://github.com/facebook/docusaurus/edit/main/docs/",
          "path": "../simple_website/docs"
        },
        "blog": {},
        "theme": {
          "customCss": "../simple_website/src/css/customTheme.css"
        },
        "googleAnalytics": {
          "trackingID": "UA-44373548-31"
        }
      }
    ]
  ],
  "plugins": [],
  "themeConfig": {
    "navbar": {
      "title": "Docusaurus",
      "logo": {
        "src": "img/docusaurus.svg"
      },
      "items": [
        {
          "to": "docs/installation",
          "label": "Docs",
          "position": "left"
        },
        {
          "to": "docs/tutorial-setup",
          "label": "Tutorial",
          "position": "left"
        },
        {
          "to": "/users",
          "label": "Users",
          "position": "left"
        },
        {
          "href": "https://github.com/facebook/docusaurus",
          "label": "GitHub",
          "position": "left"
        }
      ]
    },
    "image": "img/docusaurus.png",
    "footer": {
      "links": [
        {
          "title": "Community",
          "items": [
            {
              "label": "Twitter",
              "to": "https://twitter.com/docusaurus"
            }
          ]
        }
      ],
      "copyright": "Copyright © 2022 Facebook Inc.",
      "logo": {
        "src": "img/docusaurus_monochrome.svg"
      }
    },
    "algolia": {
      "apiKey": "3eb9507824b8be89e7a199ecaa1a9d2c",
      "indexName": "docusaurus",
      "algoliaOptions": {
        "facetFilters": [
          "language:LANGUAGE",
          "version:VERSION"
        ]
      }
    }
  }
}",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/package.json",
    "{
  "name": "docusaurus-1-website",
  "version": "2.0.0-alpha.58",
  "private": true,
  "scripts": {
    "start": "docusaurus start",
    "build": "docusaurus build",
    "publish-gh-pages": "docusaurus-publish",
    "examples": "docusaurus-examples",
    "write-translations": "docusaurus-write-translations",
    "docusaurus-version": "docusaurus-version",
    "rename-version": "docusaurus-rename-version",
    "crowdin-upload": "crowdin --config ../crowdin.yaml upload sources --auto-update -b master",
    "crowdin-download": "crowdin --config ../crowdin.yaml download -b master",
    "swizzle": "docusaurus swizzle",
    "deploy": "docusaurus deploy",
    "docusaurus": "docusaurus"
  },
  "dependencies": {
    "@docusaurus/core": "<CURRENT_VERSION>",
    "@docusaurus/preset-classic": "<CURRENT_VERSION>",
    "clsx": "^1.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/migrated_simple_site/src/css/customTheme.css",
    ":root{
  --ifm-color-primary-lightest: #3CAD6E;
  --ifm-color-primary-lighter: #359962;
  --ifm-color-primary-light: #33925D;
  --ifm-color-primary: #2E8555;
  --ifm-color-primary-dark: #29784C;
  --ifm-color-primary-darker: #277148;
  --ifm-color-primary-darkest: #205D3B;
}
",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/simple_website/docs/api-commands.md",
    "---
id: commands
title: CLI Commands
---
## Doc
",
  ],
  [
    "<PROJECT_ROOT>/packages/docusaurus-migrate/src/__tests__/__fixtures__/simple_website/docs/api-doc-markdown.md",
    "---
id: doc-markdown
title: Markdown Features
---
## Doc
",
  ],
]
`;
